doc/platform_api/cmis_management_state_machine.md

```diff
@@ -0,0 +1,138 @@
+# CMIS management state machine HLD
+
+# High Level Design Document
+## Rev 0.1
+
+# Table of Contents
+- [1. Revision](#1-revision)
+- [2. Scope](#2-scope)
+- [3. Motivation](#3-motivation)
+- [4. Abbreviations](#4-abbreviations)
+- [5. Introduction](#5-introduction)
+- [6. HLD design](#6-hld-design)
+  - [6.1 The Requirement](#61-the-requirement)
+  - [6.2 CMIS Management Platform Support](#62-cmis-management-platform-support)
+  - [6.3 Modules Management Thread](#63-modules-management-thread)
+  - [6.4 General Module Detection Flow](#64-general-module-detection-flow)
+  - [6.5 Module Detection Flow States And Description](#65-module-detection-flow-states-and-description)
+  - [6.6 Module Detection Flow States And Functions](#66-module-detection-flow-states-and-functions)
+
+# 1. Revision
+| Rev | Date | Author | Change description |
+|:----------:|:----------:|:--------------:|:----------------------:|
+| 0.1 | 23/10/2023 | Doron Barashi | Initial version|
+| | | | |
+
+# 2. Scope
+This document provides high level design for CMIS management state machine flow
+
+# 3. Motivation
+Customers want to have complete flexibility and independence between switch vendors and optical cables.
+The goal of this feature is to allow them to take our switches and easily use them interchangeably with other vendors modules, and vice versa.
+Hence, a decision has been made to transition the management of CMIS modules from FW to software SW.
+
+# 4. Abbreviations
+
+| Term | Meaning |
+|:--------:|:---------------------------------------------:|
+| SONiC| Software for Open Networking in the Cloud|
+| CMIS | Common Management Interface Specification|
+| MDF | Module Detection Flow|
+| SW | Software |
+| FW | Firmware |
+| Independent module|SW controlled|
+
+# 5. Introduction
+
+Currently, on a switch the low level FW handles all ports and cables transcievers:

```

Comment:
there is a repeat of the motivation. remove it.
the below is extension to an existing flow by the CMIS management as there are some modules which are not CMIS based and are not yet supported by the host.
in such cases it is crucial for some platform vendors to differentiate the flow if it should be managed by the sonic cmis module management running on the host or by the vendor specific code (aka FW).
In such cases and before the the module is being configured 

then cont with explaining your part.